src/pickup.cpp

```diff
@@ -24,181 +24,203 @@ int Pickup::interact_with_vehicle( vehicle *veh, const tripoint &pos, int veh_ro
 {
     bool from_vehicle = false;
 
-    int k_part = 0;
-    int wtr_part = 0;
-    int w_part = 0;
-    int craft_part = 0;
-    int cargo_part = 0;
-    int chempart = 0;
-    int ctrl_part = 0;
+    int k_part = -1;
+    int wtr_part = -1;
+    int w_part = -1;
+    int craft_part = -1;
+    int cargo_part = -1;
+    int chempart = -1;
+    int ctrl_part = -1;
+    int purify_part = -1;
     std::vector<std::string> menu_items;
     std::vector<uimenu_entry> options_message;
     const bool has_items_on_ground = g->m.sees_some_items( pos, g->u );
     const bool items_are_sealed = g->m.has_flag( "SEALED", pos );
 
-    if( veh ) {
-        k_part = veh->part_with_feature(veh_root_part, "KITCHEN");
-        wtr_part = veh->part_with_feature(veh_root_part, "FAUCET");
-        w_part = veh->part_with_feature(veh_root_part, "WELDRIG");
-        craft_part = veh->part_with_feature(veh_root_part, "CRAFTRIG");
-        chempart = veh->part_with_feature(veh_root_part, "CHEMLAB");
-        cargo_part = veh->part_with_feature(veh_root_part, "CARGO", false);
-        ctrl_part = veh->part_with_feature(veh_root_part, "CONTROLS");
-        from_vehicle = veh && cargo_part >= 0 && !veh->get_items(cargo_part).empty();
-        const bool can_be_folded = veh->is_foldable();
-        const bool is_convertible = (veh->tags.count("convertible") > 0);
-        const bool remotely_controlled = g->remoteveh() == veh;
-
-        menu_items.push_back(_("Examine vehicle"));
-        options_message.push_back(uimenu_entry(_("Examine vehicle"), 'e'));
-
-        if (ctrl_part >= 0) {
-            menu_items.push_back(_("Control vehicle"));
-            options_message.push_back(uimenu_entry(_("Control vehicle"), 'v'));
-        }
+    if( veh == nullptr ) {
+        return -1;
+    }
 
-        if( from_vehicle ) {
-            menu_items.push_back(_("Get items"));
-            options_message.push_back(uimenu_entry(_("Get items"), 'g'));
-        }
+    k_part = veh->part_with_feature(veh_root_part, "KITCHEN");
```

Comment:
What about moving the declaration of the *_part variables here where they are initialized? Could also make them const and would avoid the completely useless initialization to 0, -1 or whatever, the value is overridden anyway.